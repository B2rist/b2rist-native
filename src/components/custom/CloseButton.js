import React, {useCallback} from 'react';
import {StyleSheet, View} from 'react-native';
import {IconButton} from 'react-native-paper';
import {useNavigate} from 'react-router-native';
import PropTypes from 'prop-types';

const CloseButton = ({path}) => {
  const navigate = useNavigate();

  const close = useCallback(() => {
    navigate(path);
  }, [navigate, path]);

  return (
    <View style={styles.closeButton}>
      <IconButton icon="close" size={50} onPress={close} />
    </View>
  );
};

const styles = StyleSheet.create({
  closeButton: {
    position: 'absolute',
    left: 1,
    top: 1,
    zIndex: 10001,
  },
});

export default CloseButton;

CloseButton.propTypes = {
  path: PropTypes.string,
};

CloseButton.defaultProps = {
  path: '/',
};
