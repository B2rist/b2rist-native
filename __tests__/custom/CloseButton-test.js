import 'react-native';
import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import CloseButton from 'components/custom/CloseButton';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-native', () => ({
  ...jest.requireActual('react-router-native'),
  useNavigate: () => mockedUsedNavigate,
}));

it('close button test', () => {
  const {getByTestId} = render(<CloseButton path="/points" />);
  expect(screen.toJSON()).toMatchSnapshot();
  fireEvent.press(getByTestId('close-button'));
  expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
  expect(mockedUsedNavigate).toHaveBeenCalledWith('/points');
});
