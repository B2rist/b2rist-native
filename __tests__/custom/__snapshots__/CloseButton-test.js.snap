// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close button test 1`] = `
<View
  style={
    {
      "left": 1,
      "position": "absolute",
      "top": 1,
      "zIndex": 10001,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "end": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": undefined,
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 33,
            "borderWidth": 0,
            "elevation": 0,
            "height": 66,
            "margin": 6,
            "overflow": "hidden",
            "width": 66,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessibilityTraits="button"
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 33,
                },
              ],
            ]
          }
          testID="close-button"
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 50,
                },
                [
                  {
                    "lineHeight": 50,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ó°…–
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
